test/adapter/json_api/relationship_test.rb

```diff
@@ -0,0 +1,173 @@
+require 'test_helper'
```

Comment:
I added this file to add proper test for each relationship feature. I think some features are currently missing (e.g. defining the relationship using the block return value) and will add them ASAP.